staging/src/k8s.io/client-go/tools/cache/mutation_cache.go

```diff
@@ -117,11 +117,11 @@ func (c *mutationCache) GetByKey(key string) (interface{}, bool, error) {
 // ByIndex returns the newer objects that match the provided index and indexer key.
 // Will return an error if no indexer was provided.
 func (c *mutationCache) ByIndex(name string, indexKey string) ([]interface{}, error) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
 	if c.indexer == nil {
```

Comment:
this is a programming error, and is not expected to be encountered in a normal case. moving the lock below the nil check doesn't gain us notable performance benefit compared to the rest of the method, and makes the method harder to reason about (locking and deferring an unlock as the first step is very easy to understand)